packages/bar/src/BarTotals.tsx

```diff
@@ -0,0 +1,154 @@
+import { AnyScale, ScaleBand } from '@nivo/scales'
+import { defaultProps } from './props'
+import { BarCommonProps, BarDatum, ComputedBarDatum } from './types'
+
+interface BarTotalsProps<RawDatum extends BarDatum> {
+    bars: ComputedBarDatum<RawDatum>[]
+    xScale: ScaleBand<string> | AnyScale
+    yScale: ScaleBand<string> | AnyScale
+    layout?: BarCommonProps<RawDatum>['layout']
+    groupMode?: BarCommonProps<RawDatum>['groupMode']
+}
+
+export const BarTotals = <RawDatum extends BarDatum>({
+    bars,
+    xScale,
+    yScale,
+    layout = defaultProps.layout,
+    groupMode = defaultProps.groupMode,
+}: BarTotalsProps<RawDatum>) => {
```

Comment:
Seeing both implementations (canvas + SVG), it seems that the computation is quite similar, I think we could compute the totals in some kind of hook (`useBarTotals` or something similar), and then we would only care about rendering for each implementation. What do you think?

Reply:
Done! Great tip, thanks